File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.java
// Last updated: 8/2/2026, 1:05:34 AM
class Solution {
    public int removeDuplicates(int[] nums) {
        int size=nums.length;
        int c=1,k=0;
        for(int i=0,j=0;i<size&&j<size;j++)
        {
            if(nums[i]!=nums[j])
            {
                c+=1;
                nums[k++]=nums[i];
                i=j;

            }
        }
        nums[k]=nums[size-1];
        return c;
    }
}

File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.cpp
// Last updated: 8/2/2026, 1:05:29 AM
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int slow=0;
        for(int fast=1;fast<nums.size();fast++)
        {
            if(nums[slow]!=nums[fast])
            {
                slow++;
                nums[slow]=nums[fast];
            }
        }
        int c=0;
        for(int i=0;i<=slow;i++)
        {
            c+=1;
        }
        return c;
    }
};